CMakeLists.txt

```
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(ClpIrV1Decoder)
+
+# Download and extract Boost
+set(BOOST_VERSION "1.85.0")
+string(REPLACE "." "_" BOOST_ARCHIVE_VERSION_PART ${BOOST_VERSION})
+
+include(ExternalProject)
+ExternalProject_Add(
+        Boost
+        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
+        PREFIX "${CMAKE_BINARY_DIR}/boost"
+        CONFIGURE_COMMAND ""
+        BUILD_COMMAND ""
+        INSTALL_COMMAND ""
+        LOG_DOWNLOAD ON
+)
+ExternalProject_Get_Property(Boost SOURCE_DIR)
+set(BOOST_EXTRACT_DIR "${SOURCE_DIR}")
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++23")
+set(CMAKE_CXX_STANDARD 23)
```

Sorry for sharing you some outdated config in the first place. 

It seems the norm these days is to use `target_compile_features(targetName visibility cxx_std_XX)` because it sets the language standard per target, assuming we will add more targets into this CMakeLists.txt in the future like 
a `ClpIrV2Decoder`.

Let's also drop the CXX standard to 20 (instead of 23) to stay consistent with clp and our other ffi repos. 

```suggestion
target_compile_features(${BIN_NAME}
    PRIVATE
    cxx_std_20
)
```